lib/mongo/client.rb

```diff
@@ -1103,6 +1103,18 @@ def cluster_modifying?(new_options)
     # but does not check for interactions between combinations of options.
     def validate_new_options!(opts = Options::Redacted.new)
       return Options::Redacted.new unless opts
+      if opts[:read_concern]
+        given_keys = opts[:read_concern].keys
+        allowed_keys = ['level']
+        # raise an error for non user-settable options
+        if opts[:read_concern][:after_cluster_time]
+          raise Mongo::Error::OperationFailure.new('after_cluster_time is not a user-settable option')
+        end
+        # warn that options are invalid but keep it and forward to the server
+        if given_keys != allowed_keys
+          log_warn("Read concern has invalid key.")
```

Comment:
For this to be more helpful/actionable I would say at least the actual options given should be included, as well as what operation is being attempted.